build-zip.sh

```diff
@@ -38,62 +31,65 @@ function get_repo_base_url() {
   esac
 
   echo $BASE_URL
-} && export -f get_repo_base_url
+}
 
 function download_repo_index() {
-  local DL_URL="$(get_repo_base_url $1)/index.xml"
+  local DL_URL="$(get_repo_base_url "$1")/index.xml"
   local FILE_NAME="$1_index.xml"
 
-  fetch $DL_URL $FILE_NAME
-} && export -f download_repo_index
+  fetch "$DL_URL" "$FILE_NAME"
+}
 
 function xpath_exec() {
-  local INDEX_FILE=$1
+  local INDEX_FILE="$1"
   local XPATH_CMD="$2"
 
-  xmlstarlet select -t -v "$XPATH_CMD" $INDEX_FILE
-} && export -f xpath_exec
+  xmlstarlet select -t -v "$XPATH_CMD" "$INDEX_FILE" | head -1
+}
 
 ## Applications
-function get_stable_versioncode() {
-  local INDEX_FILE=$1
-  local PACKAGE_ID=$2
+function get_stable_version() {
+  local INDEX_FILE="$1"
+  local PACKAGE_ID="$2"
 
-  xpath_exec $INDEX_FILE "/fdroid/application[@id = '$PACKAGE_ID']/marketvercode"
-} && export -f get_stable_versioncode
+  xpath_exec "$INDEX_FILE" "/fdroid/application[@id = '$PACKAGE_ID']/marketversion"
```

Comment:
Loose. F-Droid's XML indexes have a `<nativecode>` tag. Why don't just using it?

Reply:
https://github.com/zaoqi/flashable-zip_microG/blob/59d66c4c6fb814dd5106162cd69a06e351ddfc04/build-zip.sh#L75

Reply:
Right. But better do that in the `get_app_filename()` function, because that's where the XPath query is build and executed.